### PR TITLE
Fix #560 - datasource segments summary shouldn't include replication.

### LINKED ISSUE #560: datasource segments summary shouldn't include replication
/datasources/[DATASOURCE]?simple appears to include replication in the segments bit of the response, and it shouldn't (this information is in the tiers)

this was an uncaught part of #438 
for instance:

```
http://10.151.79.16:8080/druid/coordinator/v1/datasources/dash_track
```

returns:
```
{
  "tiers": {
    "_default_tier": {
      "segmentCount": 12,
      "size": 568436361
    },
    "hot": {
      "segmentCount": 10,
      "size": 568436361
    }
  },
  "segments": {
    "minTime": "2014-01-28T21:00:00.000Z",
    "count": 22,
    "maxTime": "2014-05-21T20:00:00.000Z",
    "size": 1136872722
  }
}
```
note that segments.size = tiers._default_tier.size + tiers.hot.size

in this case, segments.size should equal tiers._default_tier.size and segments.count should be 12 (or 10, whatever)

### PR DESCRIPTION
**Pull request by noddi**
Fix issue #560. Now size/count in summary are taken from tier with the highest segments size.

### CODE REVIEW

File: server/src/main/java/io/druid/server/http/DatasourcesResource.java [hunk 1]
```diff
@@ -551,8 +551,10 @@ public int compare(DruidDataSource druidDataSource, DruidDataSource druidDataSou
       }
 
       // segment stats
-      totalSegmentCount += druidDataSource.getSegments().size();
-      totalSegmentSize += dataSourceSegmentSize;
+      if (dataSourceSegmentSize >= totalSegmentSize) {
```
Comment (fjy) [inline server/src/main/java/io/druid/server/http/DatasourcesResource.java]: I don't think this works if there are multiple datasources.

Comment (noddi) [inline server/src/main/java/io/druid/server/http/DatasourcesResource.java]: Correct me if I'm wrong, but method **getSimpleDatasource** aggregates segment size/count only for one datasource on each server.

Comment (fjy) [inline server/src/main/java/io/druid/server/http/DatasourcesResource.java]: @noddi yeah, looking through this code I wrote again, getSimpleDatasource iterates through all the servers and pulls out info about a specific datasource for each server. The information about segments in each tier includes replication, and the total segments includes replication as well. The dataSourceSegmentSize is tracking the total segment size on a per server basis and totalSegmentSize is tracking total segment size in the cluster. I don't fully understand the logic of the if statement.

Comment (fjy) [inline server/src/main/java/io/druid/server/http/DatasourcesResource.java]: Been digging through the code and I can't seem to find a place where we maintain a map of datasource names to a set of segments ids. 

So, I think the proper way to fix the issue to create a set with all segments ids associated with the datasource name and figure out the total number of segments and segment sizes from that data structure.

Comment (noddi) [inline server/src/main/java/io/druid/server/http/DatasourcesResource.java]: Thanks for the tips, I will try to solve this.

**Comment by noddi** [comment]
@fjy I've created two sets - one for tiers and one for total. Now tier stats shows only count/size for distinct segments across all servers in that tier. Summary stats - count/size for distinct segments across all server and tiers.

**Comment by xvrl** [comment]
@noddi marking this as incompatible so we don't forget to mention this in the release notes. As much as it is a bugfix, it's important to note that the behavior changed.

**Comment by himadrisingh001** [comment]
:+1:

**Comment by fjy** [comment]
:+1:

**Pull request merged**